packages/entities/entities-plugins/src/components/DeleteCustomPluginSchemaModal.vue

```diff
@@ -0,0 +1,205 @@
+          <p>{{ t('delete.confirmModalText2') }}</p>
+
+          <!--   <EntityDeleteModal
+      :action-pending="isDeletePending"
+      :description="t('delete.description')"
+      :entity-name="pluginToBeDeleted && (pluginToBeDeleted.instance_name || pluginToBeDeleted.name || pluginToBeDeleted.id)"
+      :entity-type="EntityTypes.Plugin"
+      :error="deleteModalError"
+      :title="t('delete.title')"
+      :visible="isDeleteModalVisible"
+      @cancel="hideDeleteModal"
+      @proceed="confirmDelete"
+    /> -->
+          <!-- TODO: just use KPrompt??? -->
+          <!-- TODO: i18n-t element instead
+            <p class="confirm-text">
+            {{ helpText.confirm_1 }} "<strong>{{ plugin?.name }}</strong>" {{ helpText.confirm_2 }}.
+          </p> -->
+          <KInput
+            v-model.trim="customPluginNameFormValue"
+            data-testid="confirmation-input"
+            type="text"
+          />
+        </form>
+      </div>
+    </template>
+    <template #action-buttons>
+      <div>
+        <KButton
+          appearance="outline"
+          class="cancel-button"
+          data-testid="delete-custom-plugin-form-cancel"
+          :disabled="isLoading"
+          @click="handleCancel"
+        >
+          {{ t('actions.cancel') }}
+        </KButton>
+        <KButton
+          v-if="isPluginSchemaInUse"
+          appearance="primary"
+          data-testid="go-to-plugins-list"
+          :to="{
+            name: 'plugins'
+            /*  params: {
+              control_plane_id: config.app.controlPlaneId
+            } */
+          }"
+        >
+          {{ t('actions.go_to_plugins') }}
+        </KButton>
+        <KButton
+          v-else
+          appearance="danger"
+          data-testid="delete-custom-plugin-form-submit"
+          :disabled="isDeleteButtonDisabled"
+          form="delete-custom-plugin-form"
+          :icon="isLoading ? 'spinner' : undefined"
+          type="submit"
+        >
+          {{ t('actions.confirm_delete') }}
+        </KButton>
+      </div>
+    </template>
+  </KPrompt>
+</template>
+
+<script setup lang="ts">
+import { ref, computed, type PropType } from 'vue'
+import composables from '../composables'
+import { useErrors } from '@kong-ui-public/entities-shared'
+
+const props = defineProps({
+  plugin: {
+    type: Object as PropType<{ name: string, id: string }>,
+    required: true,
+  },
+})
+
+const emit = defineEmits(['closed', 'proceed'])
+
+const { i18n: { t } } = composables.useI18n()
+const { getMessageFromError } = useErrors()
+
+const title = computed((): string => {
+  return isPluginSchemaInUse.value
+    ? t('delete.plugin_schema_in_use_title')
+    : t('delete.title', { name: props.plugin?.name || t('delete.custom_plugin') })
+})
+
+const isLoading = ref(false)
+const errorMessage = ref('')
+const customPluginNameFormValue = ref('')
+const isPluginSchemaInUse = ref(false)
+
+const handleCancel = (): void => {
+  emit('closed')
+}
```

Comment:
suggestion: instead of creating a function here for a one-time-use emit, can you just put the emit inline in the template?

Reply:
There is now additional logic in this function that wasn't there before